language/move-vm/runtime/src/loader.rs

```diff
@@ -593,6 +594,11 @@ impl Loader {
         self.verify_ty_args(func.type_parameters(), &type_params)
             .map_err(|e| e.finish(Location::Module(module_id.clone())))?;
 
+        if is_script_execution {
```

Comment:
I think somewhere (maybe not here) we should highlight that not all `public(script)` functions can serve as an entry point for the `ScriptFunction` transaction payload.